### ISSUE TITLE
Hidapi seems to require that it is built in source, when building static release binaries.

### ISSUE THREAD

**Issue by TheCharlatan** [monero-project/monero#4453]
When compiling with `make release-static-linux-x86_64`, the build currently fails with: https://paste.debian.net/1044459/ , a whole bunch of hidapi linking errors. To me this looks like hidapi needs to be built in tree, and then object files need to be directly linked to, like here: https://github.com/edorfaus/TEMPered/blob/master/CMakeLists.txt#L40 . 
@anonimal can you add hidapi as a submodule? You can use my depends recipes in contrib/depends/packages for a hint on how to compile hidapi and its dependencies. The hidapi repo readme also contains some hints on how to build: https://github.com/signal11/hidapi . On linux, I would strongly recommend to build hiadpi-libusb. It is enough to make just hidapi a submodule. Its eudev and libusb dependencies don't need to be included in tree, they can be linked from the host system.

**Comment by iDunk5400** [comment]
I did not have such errors building `release-static-linux-x86_64` on Ubuntu 16.04. However, I did have to edit CMakeLists.txt a bit. Does this help you get past the errors?
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 994d476..53d676f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -435,7 +435,9 @@ if(STATIC AND NOT IOS)
   endif()
 endif()

-find_package(HIDAPI)
+if(NOT STATIC)
+  find_package(HIDAPI)
+endif()
```

**Comment by iDunk5400** [comment]
The thing is, with the above diff `HIDAPI_FOUND` would never be set and binaries would not have hardware device support.

**Comment by cslashm** [comment]
I add this pacman package mingw-w64-x86_64-hidapi 0.8.0rc1-4 to get compiled

**Comment by TheCharlatan** [comment]
@cslashm I think the windows static cross compile works out of the box, same goes for macOS. It's just linux, where it seems like an in tree build is required to achieve a static binary.

**Comment by TheCharlatan** [comment]
Fixed with #4506 , luckily no in-tree source is required after all.
